rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/SecureCxfClientFactoryBuilder.java

```diff
@@ -0,0 +1,351 @@
+        if (StringUtils.isEmpty(endpointUrl) || interfaceClass == null) {
+            throw new IllegalArgumentException(
+                    "Called without a valid URL, will not be able to connect.");
+        }
+
+        JAXRSClientFactoryBean jaxrsClientFactoryBean = new JAXRSClientFactoryBean();
+        jaxrsClientFactoryBean.setServiceClass(interfaceClass);
+        jaxrsClientFactoryBean.setAddress(endpointUrl);
+        jaxrsClientFactoryBean.setClassLoader(interfaceClass.getClassLoader());
+        jaxrsClientFactoryBean.getInInterceptors().add(new LoggingInInterceptor());
+        jaxrsClientFactoryBean.getOutInterceptors().add(new LoggingOutInterceptor());
+
+        if (!CollectionUtils.isEmpty(providers)) {
+            jaxrsClientFactoryBean.setProviders(providers);
+        }
+
+        Client cxfClient = WebClient.client(jaxrsClientFactoryBean.create(interfaceClass));
+        if (cxfClient == null) {
+            throw new SecurityServiceException("Could not construct base client");
+        }
+        ClientConfiguration clientConfig = WebClient.getConfig(cxfClient);
+
+        if (!StringUtils.startsWithIgnoreCase(endpointUrl, "https")) {
+            LOGGER.warn("Cannot secure non-https connection " + endpointUrl
+                    + ", only unsecured clients will be created");
+        } else {
+            initSecurity(clientConfig, cxfClient, username, password);
+        }
+
+        if (disableCnCheck) {
+            disableCnCheck(clientConfig);
+        }
+
+        return new SecureCxfClientFactory<>(cxfClient);
+    }
+
+    private void disableCnCheck(ClientConfiguration clientConfig) throws SecurityServiceException {
+        HTTPConduit httpConduit = clientConfig.getHttpConduit();
+        if (httpConduit == null) {
+            throw new SecurityServiceException(
+                    "HTTPConduit was null for " + this + ". Unable to disable CN Check");
+        }
+
+        TLSClientParameters tlsParams = httpConduit.getTlsClientParameters();
+
+        tlsParams.setDisableCNCheck(true);
+    }
+
+    /*
+     * Add TLS and Basic Auth credentials to the underlying {@link org.apache.cxf.transport.http.HTTPConduit}
+     * This includes two-way ssl assuming that the platform keystores are configured correctly
+     */
+    private void initSecurity(ClientConfiguration clientConfig, Client cxfClient, String username,
+            String password) throws SecurityServiceException {
+
+        HTTPConduit httpConduit = clientConfig.getHttpConduit();
+        if (httpConduit == null) {
+            throw new SecurityServiceException(
+                    "HTTPConduit was null for " + this + ". Unable to configure security.");
+        }
+
+        if (StringUtils.isNotEmpty(username) && StringUtils.isNotEmpty(password)) {
+            String asciiString = cxfClient.getBaseURI().toASCIIString();
+            if (!StringUtils.startsWithIgnoreCase(asciiString, "https")) {
```

Comment:
I believe a check similar to this was already done before calling this method...unless I am missing when a base uri would be different than the endpointurl. Can either remove the check that calls this method or remove this check to simplify the code.